src/tests/ftest/launch.py

```diff
@@ -1459,886 +1108,1972 @@ def ignore_unknown(self, node):  # pylint: disable=no-self-use,unused-argument
+    return 0
+
+
+class LaunchException(Exception):
+    """Exception for launch.py execution."""
+
+
+class AvocadoInfo():
+    """Information about this version of avocado."""
+
+    def __init__(self):
+        """Initialize an AvocadoInfo object.
+
+        Raises:
+            LaunchException: _description_
+        """
+        try:
+            # pylint: disable=import-outside-toplevel
+            from avocado.core.version import MAJOR, MINOR
+            self.major = int(MAJOR)
+            self.minor = int(MINOR)
+
+        except ModuleNotFoundError:
+            # Once lightweight runs are using python3-avocado, this can be removed
+            log = logging.getLogger(__name__)
+            result = run_local(log, ["avocado", "-v"], check=True)
+            try:
+                version = re.findall(r"(\d+)\.(\d+)", result.stdout)[0]
+                self.major = int(version[0])
+                self.minor = int(version[1])
+            except IndexError as error:
+                raise LaunchException("Error extracting avocado version from command") from error
+
+    def __str__(self):
+        """Get the avocado version as a string.
+
+        Returns:
+            str: the avocado version
+
+        """
+        return f"Avocado {str(self.major)}.{str(self.minor)}"
+
+    def get_setting(self, section, key, default=None):
+        """Get the value for the specified avocado setting.
+
+        Args:
+            section (str): avocado setting section name
+            key (str): avocado setting key name
+            default (object): default value to use if setting is undefined
+
+        Raises:
+            LaunchException: if there is an error getting the setting from the avocado command
+
+        Returns:
+            object: value for the avocado setting or None if not defined
+
+        """
+        try:
+            # pylint: disable=import-outside-toplevel
+            from avocado.core.settings import settings
+            try:
+                # Newer versions of avocado use this approach
+                config = settings.as_dict()
+                return config.get(".".join([section, key]))
+
+            except AttributeError:
+                # Older version of avocado, like 69LTS, use a different method
+                # pylint: disable=no-member
+                try:
+                    return settings.get_value(section, key)
+                except settings.SettingsError:
+                    # Setting not found
+                    pass
+
+            except KeyError:
+                # Setting not found
+                pass
+
+        except ModuleNotFoundError:
+            # Once lightweight runs are using python3-avocado, this can be removed
+            log = logging.getLogger(__name__)
+            result = run_local(log, ["avocado", "config"], check=True)
+            try:
+                return re.findall(rf"{section}\.{key}\s+(.*)", result.stdout)[0]
```

Comment:
`re.search` is similar, but only returns the first occurrence

Reply:
The return types are different.